plugins/rapid7_insightidr/plugin.spec.yaml

```diff
@@ -469,8 +578,247 @@ actions:
         description: The pagination parameters used to generate this page result
         type: investigation_metadata
         required: true
-
-
+  create_investigation:
+    title: Create Investigation
+    description: Allows to create investigation manually
+    input:
+      title:
+        title: Title
+        description: Investigation's title
+        type: string
+        required: true
+        example: Example Title
+      status:
+        title: Status
+        description: Investigation's status
+        type: string
+        enum:
+          - ""
+          - OPEN
+          - CLOSED
+        required: false
+        example: OPEN
+      priority:
+        title: Priority
+        description: Investigation's priority
+        type: string
+        enum:
+          - ""
+          - LOW
+          - MEDIUM
+          - HIGH
+          - CRITICAL
+        required: false
+        example: LOW
+      disposition:
+        title: Disposition
+        description: Investigation's disposition
+        type: string
+        enum:
+          - ""
+          - BENIGN
+          - MALICIOUS
+          - NOT_APPLICABLE
+        required: false
+        example: BENIGN
+      email:
+        title: Email
+        description: A user's email address for investigation to be assigned
+        type: string
+        required: false
+        example: user@example.com
+    output:
+      investigation:
+        title: Investigation
+        description: The body of the specified investigation
+        type: investigation
+        required: true
+  search_investigation:
+      title: Search Investigation
+      description: Allows to search for investigations that match the given criteria
+      input:
+        search:
+          title: Search
+          description: The criteria for which entities to return
+          type: "[]object"
+          required: false
+          example: '[{"field": "Example Field", "operator": "EQUALS", "value": "Test"}]'
+        sort:
```

Comment:
Could this maybe be split up into two separate ENUM inputs?

Reply:
Hm, I am not sure. Do you mean to split up the `[]object` type into separate ENUM inputs? That may be hard to do so because of there may be many dictionaries put there as search criteria 😃